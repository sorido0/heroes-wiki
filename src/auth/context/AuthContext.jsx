import { createContext } from "react";

// Context es para poder acceder a los valores del usuario desde cualquier parte de nuestra aplicacion
export const AuthContext = createContext();