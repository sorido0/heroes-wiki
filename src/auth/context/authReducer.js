
import { types } from './../types/types';
// Es el valor inicial de nuestro del reducer de auth
const estadoInicial = {
    logged: false,
}

export const authReducer = (state = estadoInicial, action) => {
   // console.log(" Tipo de accion desde el test+++++++++ " + action.type)

   

    //console.log("esto es action                  :" + action.type);
    //console.log("esto es type.logout                  :" + types.logout);
    //console.log("esto es action", action.type);

    switch (action.type) {
        // Si el tipo de acci√≥n es login, entonces...
        case types.login:
            return {
                ...state,
                logged: true,
                user: action.payload
            }

        case types.logout:
            return {
                ...state,
                logged: false,
            }

        default:
            return state
    }
}