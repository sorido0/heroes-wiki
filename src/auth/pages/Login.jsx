
import { useNavigate } from 'react-router-dom';

export const Login = () => {

    const enHome = useNavigate();

    const eLogin = () => {
        enHome('/', {
            replace: true
            });
    }
    return (
        <div className='container mt-5'>
            <h1 
            className='p-5 text-center'
            >   
                Login
            </h1>
            <hr />

            <button 
            className="btn btn-primary d-flex justify-content-center m-5"
            to="/Login"
            onClick={eLogin}

            > Entrar </button>
        </div>
    )
};
