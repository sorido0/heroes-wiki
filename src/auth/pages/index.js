



export * from './Login';
export * from '../context';