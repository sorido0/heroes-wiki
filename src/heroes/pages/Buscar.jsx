

export const Buscar = () => {
    return (
        <>
            <label htmlFor="buscar">
                <input
                    type="search"
                    name="buscar"
                    id="buscar"
                    placeholder='Buscar heroe' 
                    data-tesrin='label-inp-btn'/>
            </label>
        </>
    )
}
