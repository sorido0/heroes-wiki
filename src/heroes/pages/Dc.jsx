
import { InfoHeores } from './../components/InfoHeores';

export const Dc = () => {
    return (
        <div className='container'>
            <h1>Dc</h1>
            <hr />
            <InfoHeores value='DC Comics'/>
        </div>

    )
};
