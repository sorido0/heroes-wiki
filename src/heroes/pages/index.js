


export * from './Dc';
export * from './Mavel';
export * from './Heroe';
export * from './Buscar';