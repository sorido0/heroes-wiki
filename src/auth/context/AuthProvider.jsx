
import { useReducer } from 'react';
//import { types } from '../types/Types';
import { AuthContext } from './AuthContext';
import { authReducer } from './authReducer';


const estadoInicial = { }

const types = {
    login: 'login',
    logout: 'logout',
}

const init = () => {
    const user = JSON.parse(localStorage.getItem('user'));

    return {
        logged: !!user,
        user
    }
}


export const AuthProvider = ({ children }) => {

    
    // importamos el reducer de auth
    const [state, dispatch] = useReducer(authReducer, estadoInicial, init);

    // Creamos la funcion onLogin
    const onLogin = (name = 'mmg') => {

        const user = { id: 'yunito' , name };

        const action = {
            type: types.login,
            payload: user
        }

        localStorage.setItem('user', JSON.stringify(user));

        dispatch(action)
    }

    const onLogout = () => {
        localStorage.removeItem('user');

        const action = {
            type: types.logout
        }
        dispatch(action)
    }

    return (
        // El provider es un componente que se encarga de proveer el valor de un contexto
        <AuthContext.Provider value={{
            ...state,
            onLogin,
            onLogout 
        }}>
            {/* Los children los las informacion que que recive el provider */}
            {children}
        </AuthContext.Provider >

    );
}
