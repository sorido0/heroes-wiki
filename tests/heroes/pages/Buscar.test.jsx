
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from './../../../src/auth/context/AuthContext';
import { Buscar } from './../../../src/heroes/pages/Buscar';



describe('Pruebas en el componente <Buscar>', () => { 


    test('Debe de mostrar los valores por defectos ', () => { 

        
        const { container } = render(
            //<AuthContext.Provider>
                <MemoryRouter>
                <Buscar />
                </MemoryRouter>
            //</AuthContext.Provider>
            
        )

        //screen.debug();
        expect( container ).toMatchSnapshot();

     })

     test('De debe mostrar a Batman y el input con el valor el queryString ', () => { 

        render(
            <MemoryRouter initialEntries={['/buscar?q=batman']}>
                <Buscar />
            </MemoryRouter>
        )
            
            // para representar el valor de cualquier elemento del DOM
            const ima = screen.getByRole('img');
            
            //screen.debug();
            expect( screen.getAllByText('Batman').length  ).toBeGreaterThanOrEqual(1);
            // toContain Es para buscar un texto dentro de un elemento
            expect( ima.src ).toContain('src/assets/heroes/dc-batman.jpg');
      })

 })