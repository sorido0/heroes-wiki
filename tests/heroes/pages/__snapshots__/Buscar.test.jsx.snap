// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Buscar> Debe de mostrar los valores por defectos  1`] = `
<div>
  <div
    class="container"
  >
    <h1>
       Buscar heroe 
    </h1>
    <br />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <h4>
           Buscar 
        </h4>
        <hr />
        <form>
          <input
            autocomplete="off"
            class="form-control"
            id="BuscarText"
            name="BuscarText"
            placeholder="Buscar heroe"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary mt-2"
            type="submit"
          >
            Buscar
          </button>
        </form>
      </div>
      <div
        class="col-7"
      >
        <h4>
           Resultados 
        </h4>
        <hr />
        <div
          class="alert alert-primary animate__animated animate__fadeIn"
        >
          <p>
             ðŸ˜Ž Realiza una busqueda 
            <span>
               ... 
            </span>
             
          </p>
        </div>
        <div
          class="container"
        />
      </div>
    </div>
  </div>
</div>
`;
